src/metadata/metadata_passive_io.c

```diff
@@ -0,0 +1,377 @@
+			continue;
+
+		_dec_core_stats(core);
+
+		ocf_hb_cline_prot_lock_wr(&cache->metadata.lock, lock_idx, core_id,
+				core_line);
+		ocf_metadata_remove_from_collision(cache, cline, PARTITION_DEFAULT);
+		ocf_hb_cline_prot_unlock_wr(&cache->metadata.lock, lock_idx, core_id,
+				core_line);
+	}
+}
+
+static int passive_io_resume(struct ocf_request *req)
+{
+	ocf_cache_t cache = req->cache;
+	struct ocf_metadata_ctrl *ctrl = cache->metadata.priv;
+	struct ocf_io *io = (struct ocf_io*) req->data;
+	ctx_data_t *data = ocf_io_get_data(io);
+	uint64_t io_start_page = BYTES_TO_PAGES(io->addr);
+	uint64_t io_end_page = io_start_page + BYTES_TO_PAGES(io->bytes);
+	ocf_end_io_t io_cmpl = req->master_io_req;
+	enum ocf_metadata_segment_id update_segments[] = {
+		metadata_segment_sb_config,
+		metadata_segment_part_config,
+		metadata_segment_core_config,
+		metadata_segment_core_uuid,
+		metadata_segment_collision,
+	};
+	int i;
+
+	for (i = 0; i < ARRAY_SIZE(update_segments); i++) {
+		ocf_cache_line_t cache_line_range_end, cache_line_range_start;
+		enum ocf_metadata_segment_id seg = update_segments[i];
+		struct ocf_metadata_raw *raw = &(ctrl->raw_desc[seg]);
+		uint64_t raw_start_page = raw->ssd_pages_offset;
+		uint64_t raw_end_page = raw_start_page + raw->ssd_pages;
+		uint64_t overlap_start = OCF_MAX(io_start_page, raw_start_page);
+		uint64_t overlap_end = OCF_MIN(io_end_page, raw_end_page);
+		uint64_t overlap_start_data = overlap_start - io_start_page;
+		uint64_t overlap_page;
+		uint64_t overlap_count;
+
+		if (overlap_start >= overlap_end)
+			continue;
+
+		overlap_page = overlap_start - raw_start_page;
+		overlap_count = overlap_end - overlap_start;
+
+		if (seg == metadata_segment_collision) {
+			/* The collision is not updated yet but the range of affected
+			   cachelines is already known. Remove the old mapping related info
+			   from the metadata */
+			/* The range of cachelines with updated collision section */
+			cache_line_range_start = overlap_page * raw->entries_in_page;
+			cache_line_range_end = cache_line_range_start +
+				raw->entries_in_page * overlap_count;
+
+			cleanup_old_mapping(cache, cache_line_range_start,
+					cache_line_range_end);
+		}
+
+		ctx_data_seek(cache->owner, data, ctx_data_seek_begin,
+				PAGES_TO_BYTES(overlap_start_data));
+		ocf_metadata_raw_update(cache, raw, data, overlap_page, overlap_count);
+
+		if (seg != metadata_segment_collision)
+			continue;
+
+		update_list_segment(cache, cache_line_range_start,
+				cache_line_range_end);
+	}
+
+	ocf_pio_async_unlock(req->cache->standby.concurrency, req);
+	io_cmpl(io, 0);
+	env_allocator_del(cache->standby.allocator, req);
+	return 0;
+}
+
+static struct ocf_io_if passive_io_restart_if = {
+	.read = passive_io_resume,
+	.write = passive_io_resume,
+};
+
+static void passive_io_page_lock_acquired(struct ocf_request *req)
+{
+	ocf_engine_push_req_front(req, true);
+}
+
+int ocf_metadata_passive_update(ocf_cache_t cache, struct ocf_io *io,
```

Comment:
I think the file name is misleading, there is no I/O in this file (metadata_passive_io.c)

Reply:
I renamed it to metadata_passive_update.c